libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/LoginContext.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+@objc(SFSDKLoginContext)
+@objcMembers
+class LoginContext: NSObject {
+    var loginHint: String?
+    var loginHost: String?
+    var manager: UserAccountManaging?
+    var appConfig: BootstrappingConfig?
+
+    init(loginHint: String?, loginHost: String?, userAccountManager: UserAccountManaging?, appConfig: BootstrappingConfig?) {
+        self.loginHint = loginHint
+        self.loginHost = loginHost
+        self.manager = userAccountManager
+        self.appConfig = appConfig
+    }
+
+    func hasLoginHint() -> Bool {
+        return (loginHint?.count ?? 0) > 0
+    }
+
+    func hasLoginHost() -> Bool {
+        return (loginHost?.count ?? 0) > 0
+    }
+
+    func shouldLogin() -> Bool {
```

Comment:
@wmathurin @bbirman  I add this method for checking if login should happen.  Added the context of has a hint or a host.